src/keychain_mac.cpp

```diff
@@ -101,67 +102,80 @@ void updateError(keychain::Error &err, OSStatus status) {
     }
 }
 
-void handleCFCreateFailure(keychain::Error &err,
-                           const std::string &errorMessage) {
+void setGenericError(keychain::Error &err, const std::string &errorMessage) {
+    err = keychain::Error{};
     err.message = errorMessage;
     err.type = keychain::ErrorType::GenericError;
     err.code = -1;
 }
 
-CFStringRef createCFStringWithCString(const std::string &str,
-                                      keychain::Error &err) {
-    CFStringRef result = CFStringCreateWithCString(
-        kCFAllocatorDefault, str.c_str(), kCFStringEncodingUTF8);
-    if (result == NULL) {
-        handleCFCreateFailure(err, "Failed to create CFString");
+/*! \brief Helper to manage the lifetime of CFObjects */
+template <typename T> struct ScopedCFRef {
```

Comment:
```suggestion
template <typename T, typename = typename std::enable_if<std::is_pointer<T>::value>::type>
class ScopedCFRef final {
```

If you want to control member access you might as well make this a class.
Also: `CFRef`s are pointers and your class depends on this fact. The `enable_if` at least makes sure that `T` is a pointer.